### DISCUSSION TITLE
About set PDF Filter

### DISCUSSION THREAD

**Discussion by Syntamin** [pymupdf/PyMuPDF#3671] [Q&A]
### Description of the bug

About set PDF Filter

### How to reproduce the bug

Hello,

During my process of compressing PDFs using PyMuPDF, I noticed that after converting images to JPEG, the original object's Filter is set to /DCTDecode. However, when I used other online tools to compress PDFs, I observed that they seem to employ two layers of compression: one with JPEG compression (setting the Quality to 60 in DecodeParms), and another with the FlateDecode filter. This approach seems to significantly reduce the PDF size.

I would like to know if it is possible to achieve this using PyMuPDF. If so, after setting the Filter entry for the object, what additional steps are required to handle the content stream properly?

Thank you for your assistance.

Best regards,
Syntamin

### PyMuPDF version

1.24.7

### Operating system

MacOS

### Python version

3.11

**Comment by Syntamin** [comment]
This is an object from a PDF compressed using an online tool:
```24 0 obj
<<
/Type /XObject
/Subtype /Image
/Width 1232
/Height 1280
/BitsPerComponent 8
/ColorSpace /DeviceGray
/Filter [/FlateDecode /DCTDecode]
/DecodeParms [null <<
/Quality 60
>>]
/Length 13389
>>
stream
```

And this is my an object:
```
3 0 obj
<</Type/XObject/Subtype/Image/Length 310642/Width 1280/Height 1280/BitsPerComponent 8/SMask null/ColorSpace/DeviceRGB/Filter/DCTDecode>>
stream
```

**Comment by JorjMcKie** [comment]
No, MuPDF does not support this.
This is no issue but a Discussions item ... transferring.
